src/controller.rs

```diff
@@ -521,19 +584,27 @@ mod tests {
             // `get_valid_credentials()` returns `Err`
             mock_manager
                 .expect_get_valid_credentials()
-                .return_once(move |_, _, _| Err("".into()));
+                .return_once(move |_, _, _| {
+                    Err(CliError::CredentialError(
+                        CredentialError::InvalidCredentials,
+                    ))
+                });
 
-            let controller = Controller {
+            let mut controller = Controller {
                 api: mock_api,
                 exit_fn: |_| {},
                 manager: mock_manager,
                 is_loop: true,
                 is_boot: true,
+                show_notif_fn: |_: &str| {},
             };
-            let result = controller.run_auto_sign_up(|_| {}).await;
+            let result = controller.run_auto_sign_up().await;
 
             assert!(result.is_err());
-            assert_eq!(result.unwrap_err(), "Invalid credentials");
+            match result.unwrap_err() {
+                CliError::CredentialError(CredentialError::InvalidCredentials) => {}
+                _ => panic!("Expected InvalidCredentials error"),
+            }
```

Comment:
Shouldn't you be using assertEq!()?

Reply:
Have to make the enum derive PartialEq then